### PR TITLE
[ADD] website_sale_order_company

### PR DESCRIPTION
**Pull request by eLBati**
# e-commerce order company

This module sets online (ecommerce) order's company according to purchased products.
# Usage

Set 'company' field for products that must be sold by that company.
If a product can be sold by any company, just leave company field empty.

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Can you explain the reasoning behind this module?

**Comment by eLBati** [comment]
@pedrobaeza updated https://github.com/eLBati/website/blob/website_sale_order_company/website_sale_order_company/README.rst
See if it's clearer

**Comment by pedrobaeza** [comment]
OK, I see. It's a little twist to get a multi-company shop, but it can serve until v9 arrives.

File: website_sale_order_company/models/sale_order.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        Check products companies. If all products have the same company
+        (or empty), return it
+        """
+        companies = {}
+        company = False
+        for order in self:
+            for line in order.order_line:
+                if line.product_id and line.product_id.company_id:
+                    companies[line.product_id.company_id] = True
+        if len(companies) == 1:
```
Comment (pedrobaeza) [inline website_sale_order_company/models/sale_order.py:42]: Shouldn't you return companies[0]?, because a sale order cannot be without company.

Comment (eLBati) [inline website_sale_order_company/models/sale_order.py:42]: If `get_products_company` doesn't find a unique company, the default one id used (returned by '_get_default_company' of 'sale.order')

Also updated comment

**Comment by pedrobaeza** [comment]
Thanks :+1:

**Comment by hbrunn** [comment]
:+1:

**Pull request merged**